libraries/chain/kv_context_rocksdb.cpp

```diff
@@ -0,0 +1,315 @@
+         try {
+            try {
+               return b1::chain_kv::compare_key(kv_it.get_kv(), b1::chain_kv::key_value{ { key, size }, {} });
+            }
+            FC_LOG_AND_RETHROW()
+         }
+         CATCH_AND_EXIT_DB_FAILURE()
+      }
+
+      kv_it_stat kv_it_move_to_end() override {
+         try {
+            try {
+               kv_it.move_to_end();
+               return kv_it_stat::iterator_end;
+            }
+            FC_LOG_AND_RETHROW()
+         }
+         CATCH_AND_EXIT_DB_FAILURE()
+      }
+
+      kv_it_stat kv_it_next(uint32_t* found_key_size, uint32_t* found_value_size) override {
+         EOS_ASSERT(!kv_it.is_erased(), kv_bad_iter, "Iterator to erased element");
+         try {
+            try {
+               ++kv_it;
+               get_current_key_value_sizes(found_key_size, found_value_size);
+            }
+            FC_LOG_AND_RETHROW()
+         }
+         CATCH_AND_EXIT_DB_FAILURE()
+
+         return kv_it_status();
+      }
+
+      kv_it_stat kv_it_prev(uint32_t* found_key_size, uint32_t* found_value_size) override {
+         EOS_ASSERT(!kv_it.is_erased(), kv_bad_iter, "Iterator to erased element");
+         try {
+            try {
+               --kv_it;
+               get_current_key_value_sizes(found_key_size, found_value_size);
+            }
+            FC_LOG_AND_RETHROW()
+         }
+         CATCH_AND_EXIT_DB_FAILURE()
+
+         return kv_it_status();
+      }
+
+      kv_it_stat kv_it_lower_bound(const char* key, uint32_t size, uint32_t* found_key_size, uint32_t* found_value_size) override {
+         try {
+            try {
+               kv_it.lower_bound(key, size);
+               get_current_key_value_sizes(found_key_size, found_value_size);
+            }
+            FC_LOG_AND_RETHROW()
+         }
+         CATCH_AND_EXIT_DB_FAILURE()
+
+         return kv_it_status();
+      }
+
+      kv_it_stat kv_it_key(uint32_t offset, char* dest, uint32_t size, uint32_t& actual_size) override {
+         EOS_ASSERT(!kv_it.is_erased(), kv_bad_iter, "Iterator to erased element");
+
+         std::optional<b1::chain_kv::key_value> kv;
+         try {
+            try {
+               kv = kv_it.get_kv();
+            }
+            FC_LOG_AND_RETHROW()
+         }
+         CATCH_AND_EXIT_DB_FAILURE()
+
+         if (kv) {
+            if (offset < kv->key.size())
+               memcpy(dest, kv->key.data() + offset, std::min((size_t)size, kv->key.size() - offset));
+            actual_size = kv->key.size();
+            return kv_it_stat::iterator_ok;
+         } else {
+            actual_size = 0;
+            return kv_it_stat::iterator_end;
+         }
+      }
+
+      kv_it_stat kv_it_value(uint32_t offset, char* dest, uint32_t size, uint32_t& actual_size) override {
+         EOS_ASSERT(!kv_it.is_erased(), kv_bad_iter, "Iterator to erased element");
+
+         std::optional<b1::chain_kv::key_value> kv;
```

Comment:
This method looks almost identical to the method above.  Would it add to much complexity to create a common method they both call?